### DISCUSSION TITLE
Open up for group discussion: Metastasis

### DISCUSSION THREAD

**Discussion by rtmill** [OHDSI/OncologyWG#477] [Ideas]
# Problem Space

Standard representation for metastasis.

# Proposed Solutions

Findings of metastasis when attributed to a cancer would be represented as modifiers (measurements) with FK back to that condition. Those modifiers would consist of a standard discrete set using the ["Metastasis" class within the Cancer Modifier Vocabulary](https://athena.ohdsi.org/search-terms/terms?vocabulary=Cancer+Modifier&conceptClass=Metastasis&page=1&pageSize=15&query=). 

One point of contention is whether or not to represent all concepts that contain evidence of metastasis solely as measurements, or also to persist them as conditions. For instance, there are many ICD codes that imply metastasis and are in the condition domain, and this is helpful when leveraging the condition hierarchies to find all forms of neoplastic disease. The proposals seem to boil down to:

Option 1.  map all of those conditions to measurements and consequentially break the condition hierarchy of neoplastic disease
Option 2.  map those condition concepts to BOTH measurements (within Cancer Modifier vocab) AND to standard condition concepts

**Comment by rtmill** [comment]
Please add any thoughts or questions!

**Reply by jam560** [reply]
FYI, DFCI curation uses ICD-O-3 sites the dictionary for sites of metastases. It contains 334 entries that appear to be a subset of the metastasis cancer modifiers. See attached "74-SITE-DM" file.  For what it's worth, we also record the "site" of the scan from which mets sites were determined, which is the short list:

Brain/Head
Spine
Neck
Chest
Abdomen
Pelvis
Extremity
Full Body

[74_SITE_DM.xlsx](https://github.com/OHDSI/OncologyWG/files/10460820/74_SITE_DM.xlsx)

**Comment by cgreich** [comment]
@rtmill:

I don't think we have an option other than 1. Reasons:

- When creating the cancer disease model, we left only histology and topology of the primary in the Condition. Shouldn't break our own model. People will be utterly confused.
- A metastasis something that happens over time to a primary, it is not a condition of its own. In fact, you can see in the histology where it came from (except when it deteriorated so badly that it looks generic basophil with large nuclei all dividing).
- There is no hierarchy between metastases and primaries. There is only hierarchy of metastatic sites. That can be done in the Measurement (they can have hierarchies).

If you want to find all neoplasms, but the data contain no primary condition but only metastases then yes, you would have to look in both tables. Not a big deal.

@jam560: Will check them out against our list.

**Reply by jam560** [reply]
Just validating in my own mind. I think the most common use cases would be:

A. Find all the metastatic sites from a given primary cancer type (and e.g. compute their different survival curves). 

B. Find all the patients with primaries that metastasized to the brain. 

C. A or B combined with treatment regimen.

I think all of those are at satisfied by option 1.

**Comment by golozara** [comment]
From research standpoint, option 1 is the most straight forward options. @jam560's use cases can be all done with option 1. If you go with option 2, we will only add confusion when it comes to phenotyping: we will go against conventions and add more complexity to something already complex.

**Comment by askanter** [comment]
What is the intent in regards to mapping the metastases to ICD and SNOMED? If a patient has "breast cancer metastasized to pelvis" which includes a primary breast malignancy and a secondary pelvic malignancy (or metastatic malignancy to pelvis)? Are we creating a separate anatomical hierarchy? It seems like I don't understand Christian's reply.

**Reply by cgreich** [reply]
We are mapping ICDs to Cancer Modifiers of the Class "Metastasis". They contain metastases to all sorts of anatomical structures. They are not linked to a primary per concept relationship. In other words, there is no pelvic metastasis of breast cancer, or pelvic metastasis of colon cancer. There is just a pelvic metastasis.

And yes, inside the metastases there is a simplified anatomical hierarchy. Works?

**Reply by askanter** [reply]
Well I was thinking about how the US (via IMO) captures cancer diagnoses... Stage 3 Malignant neoplasm of the breast, ER+, PR- or Breast cancer metastasized to pelvis... The latter is mapped to three SNOMED codes which would be available by ETL: 

94480000 | Metastatic malignant neoplasm to pelvis (disorder)
372137005 | Primary malignant neoplasm of breast (disorder)
254838004 | Carcinoma of breast (disorder)

This doesn't take into account other data which might be obtained by path reports, radiology reports, etc. from a registry. However, this data is linked in the problem... why can't we keep the fact that the breast tumor is what is in the pelvis?

**Comment by rtmill** [comment]
The decision to me boils down to what is cleaner vs. what enables a larger variety of data to be ingested and consequentially a larger variety of use cases. 

Many sites have more detailed data than is typically found in claims or (historical) EMRs - though that has changed in recent years. Specifically, when we have events related directly to the metastatic tumor, option 2 is required if we are going to use the same convention mechanisms to represent the same sorts of events. For example, measurements (sequencing, dimension, location, etc.) and treatments (surgeries, radiotherapy, etc.) that are specific to the site. Imagine a condition of "lung cancer" with associated procedure of "excision of the brain". 

I spoke with a neuro onc colleague about this and he was fairly confident we would want to have that ability to attribute events and measurements to specific metastases as, if for no other reason, they are often ultimately the main factor of the outcome (death). 
> my opinion would be that one metastasis can be very different from the next and so having separate condition events would be important. The patients can also have multiple metastasis simultaneously (often in different locations, treated differently eg with radiotherapy or not) and so they would need to be updated separately. The course that a metastasis runs is also often different with some responding to treatment and shrinking, others, not.

The tricky part is aligning this all during episode derivation, since most likely you are going to have repeated occurrences of the same diagnosis for each visit (billed for) related to treatment, and on top of that these more highly detailed representations on the same population. Taking into consideration that the data sources will vary in the level of detail and attribution of lesions, I'd vote to enable both to be integrated. 

In other words, **what is the downside of creating _both_ a measurement and a condition?** The measurement (modifier) tell you the source of the metastasis (via link to the original condition occurrence) and the condition keeps the neoplastic disease hierarchy intact as well as allows sites to attribute data, when they have that available, to the metastatic neoplasm. 

Perhaps if this isn't sufficient enough of an argument we can provide examples of what our source data looks like, in pseudo form, and rationalize how they would fit into the model?

**Comment by askanter** [comment]
This actually raised some important points. A patient might have initially a breast cancer metastasized to pelvis on the problem list, but then later also have the breast cancer mets somewhere else. It would seem that ultimately we need the measurement modifiers to manage the progress of each met, but I think it would be important to properly "decode" the pre-coordinated diagnostic terms if they do exist.

**Comment by jam560** [comment]
I would be careful of getting too complex. I might even channel Christian and say "OMOP is not your attic" and "what are the use cases?" (actual, not theoretical)? I am very doubtful that we will see any significant number of data sources in which metastatis size measurements are available. I propose we start with just capturing sites and dates of metastasis. We can always expand later when we have both use cases and commonly available data.

I think the uses cases current data sources can support are things like incidence of metastasis to specific sites pursuant to specific primaries (e.g. distribution of colon mets to liver, lung, bone, etc.).

To the question "why can't we keep the fact that the breast tumor is what is in the pelvis?" I'd answer that multiple primaries are an extreme edge case, so the source of mets is almost always implicitly the site of the primary.

**Reply by rtmill** [reply]
> I would be careful of getting too complex. I might even channel Christian and say "OMOP is not your attic" and "what are the use cases?" (actual, not theoretical)? 

Oh this is worse than I thought. 

I agree, keep it simple when we can, but there is a clear downside when loss of information relevant for research is lost because there isn't a home for it - that is, IF the goal is to make an global interoperable standard with extensibility in mind. 

For what it's worth, we have these use cases. And even further, which has at least some bearing on this conversation, another use case where there are multiple lesions at the same site (liver) with differing attributes (sequencing, dimension, topography) and treatments  - but I'll create another thread for that can of worms. The point of bringing it up is that regardless of this decision, "lesion-centric" attributes are of interest to many. 

That said, those conversations and decisions could be a later conversation for a different version/release if it's going to be a bottleneck.

**Comment by askanter** [comment]
I might agree on the details of the mets, but I actually think it is common to have more than one primary over the course of one's life. The mets do not necessarily correspond to the last identified primary. That is something that could be tested though.

**Comment by cgreich** [comment]
> Oh this is worse than I thought. .

Now, that we are in this metaphorical space, I'd called @rtmill a repeat recurrence. Can't help kicking the same dead horse. :)

When a cancer has metastasized the game is over. It's stage IV (with some deviation). Treatments become palliative. Treating metastases is in vain. It does not add a day of additional survival if you did, all you do is to make the patient suffer.

There are exceptions to this, and the medical practice will keep pushing the envelope. For some cancers with solitary metastases you can try to remove it if the primary already underwent radical treatment. And, when a metastasis causes complications (e.g. obstruct something) surgery or radiation can give relief. But the surgery already declares what anatomical structure it is about, and if you really are into studying these situations you can align them with the metastasis. However, this is not the use case we are hoping to make inroads into (survival of patients with primaries of defined stages treated in a certain way).

If you have several primaries and you don't know which one the mets belong to: This is indeed a case that happens, and we could count how often if we cared. For prognosis and treatment the above holds true though: You are in palliative space. Knowing which of the primaries metastasized might be useful to know for handling that particular patient, but for OHDSI-type studies of cohorts it is completely irrelevant. You will not find two patients with an identical situation, leave alone the 100s you need for a study.

There is a reason why cancer is a game of early detection. And there is a reason that AJCC stages are defined through M0 (no mets) and M1 (mets, no matter where).

I have empathy for what you are aiming at: A full model of the disease, beginning to end, with all the facts properly linked up. Unfortunately, this particular one has no use case for us. Plus you will not get the data to fill in your model. Diagnostic activities cease by and large at this point.

**Comment by jam560** [comment]
Part of the answer lies in our definition of "use case", which I hope to gain consensus on as part of "on ramp" documentation work under the heading of "what can you expect from OMOP Oncology?" To me (and Christian, above), it means "a research question one can reasonably expect to answer *across a network of sites* given commonly available data sources". It *doesn't* mean "finding an OMOP representation for any edge data types or associations some lucky cancer center might have available."  80/20 rule. Let's focus on enabling the most common real uses cases first, then we can iterate on deeper ones that require richer data. For example, NLP can be expected to help over time with structured data extracted from notes (e.g. response to treatment, reason for treatment discontinuation) but that data is hypothetical at this point at the network level. Let's focus on actual achievable use cases given the state of data today. At the moment, unfortunately, that's limited to incidence, lines of therapy/soc, survival (assuming everyone has decent death data), and inferred outcomes like "a new line of therapy implies recurrence or progression".

**Comment by rtmill** [comment]
@cgreich & @jam560 Your points are noted and appreciated. Focusing on the lowest hanging fruit and short-term wins has it's value and should be considered.

I'd rather not keep the discussion focused on the assertions above, though happy to elaborate if we're at an impasse otherwise. That said: yes, there are use cases that this is dependent on, yes, we do (already) have a collection of sites that have this data, yes, attributes about metastases are relevant for a plethora of scientific questions that can potentially significantly impact patient care, and regardless of the OHDSI standard study methodologies, having your data in OMOP does not confine your research solely to those approaches. 

What I'd like to focus on is the establishment of conventions of storing data if a group of folks value that data and want to conduct research with it. With the initial release of the oncology extension site after site got to a point where there was no specification of what do with their data and ended up branching off, creating ad hoc approaches and incompatible data models, which is what I'm trying to avoid. 

*To put the issue at hand succinctly:*
**Proposal** : Map non-standard metastatic condition codes to both measurements and conditions
**Pros** : 
1) Sites with attributes about non-primary lesions and have interest in that research can store it in a common way. A standardized convention, rather than an ad hoc workaround
2) It keeps the neoplastic disease hierarchy intact. A useful mechanism commonly used in phenotypes (most commonly as exclusion criteria)

**Cons** : 
1) We would need to create two mappings from non-standard metastatic condition codes (which at the moment is quite broken regardless, with some mapping to measurements, others to conditions, others to both, others to none). I'd be happy to address that if the time sink is the concern 
2) ???

I'm not proposing ripping up or revamping the fundamentals of the CDM. Even if the value of the aforementioned data is not clear to you, why prevent others from conducting research via preserving that data in a common convention, especially when the change to enable it is seemingly insignificant?

**Comment by gkennos** [comment]
I have come across this discussion when looking for how to handle a related but not precisely the same problem: what to do with diagnoses of secondary tumors that are marked as such in the clinical record, but for whatever reason have not been linked to the related primary. I recognise that for most cancer registry data, which register primaries only, this is moot, but for a clinical data source where patient records are incomplete for whatever reason, I do think this is important to handle. 

In our instance, I am using the measurement table to store measurement_concept_id=36768229 (Primary Tumor) and then just treating it as boolean, as this will at least allow filtering of cohorts depending on whether inclusion criteria refer to primaries only or all diagnosis records. Not ideal, and certainly won't be useful in collaborative settings as it stands, but just adding +1 for the measurement table solution, and petitioning for the extension to perhaps include terms for marking primary disease explicitly as well as making an affordance for locally-advanced disease that may be a secondary dx but not strictly metastatic?

**Comment by rtmill** [comment]
@mgurley Enlightened me to a fundamental flaw in my suggestion: If the data source contains attributes about a metastatic tumor, as well as the link to the primary, how would you create the relation between the two if both are stored as conditions. 

Consequently, here's relevant forum post discussing a more significant proposal: https://forums.ohdsi.org/t/fact-relationships-searching-for-an-extensible-approach/18393

**Comment by cgreich** [comment]
@gkennos:

>  I am using the measurement table to store measurement_concept_id=36768229 (Primary Tumor) and then just treating it as boolean, as this will at least allow filtering of cohorts depending on whether inclusion criteria refer to primaries only or all diagnosis records.

Not sure what you are trying to do, here. Can you explain?

In the OMOP Oncology model, all conditions are primary cancers. "Secondary" cancers are metastases. Metastases are Cancer Modifiers of primaries. You cannot have a metastasis without a primary (but you can have the situation where the primary is not detected).

So, put all primaries into CONDITION_OCCURRENCE, all mets into MEASUREMENT using Cancer Modifier concepts, and link them up (or don't, as discussed elsewhere the assignment of a met to a primary is of very little clinical value).

**Comment by rtmill** [comment]
To help with context, I did more of a dive into current metastasis mappings. Whatever decisions we end up making, I believe it's clear the current state is not consistent. 

Initial thoughts: Even if we synchronize these vocabularies into a singular hierarchical form, we're still going to need an additional qualifier on other types of concepts that are indicative of the existence of metastasis if we are going to facilitate comprehensive phenotyping.  To that end, one solution would be to also create relationships to those concepts: 

e.g. 
All concepts under: https://athena.ohdsi.org/search-terms/terms/1635142 
- have a relation of type "indicates" (or something) 
- that points to "metastasis". 


Anyway...

With the caveat of defining the "source metastasis condition concepts" as :
```
select concept_id, concept_name, domain_id, vocabulary_id, concept_code, standard_concept
	 FROM concept
 	 WHERE Concept_id in (
	    SELECT concept_id_2
	    FROM concept_relationship cr
	    WHERE concept_id_1 = 44498002 -- ICDO Neoplasm, metastatic
	    AND relationship_id = 'Histology of ICDO'
  		)
 	 union 
	  	select concept_id, concept_name, domain_id, vocabulary_id, concept_code, standard_concept
		from concept
		where vocabulary_id in ('ICD10CM', 'ICD9CM')
		and lower(concept_name) like 'secondary%'
		and lower(concept_name) like '%neoplasm%'
	union 
		select concept_id, concept_name, domain_id, vocabulary_id, concept_code, standard_concept
		from concept
		where concept_id in (
			select descendant_concept_id
			from concept_ancestor ca 
			where ancestor_concept_id = 432851 -- SNOMED: secondary malignant neoplastic disease
		)
```


<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">




**1) Counts source concepts, and the corresponding domain mapping, by vocabulary**
Takeaway: There is significant inconsistency as to which domain(s) these are mapped to. 

vocabulary_id | source_concept_count | mapped_to_condition | mapped_to_measurement
-- | -- | -- | --
ICD10CM | 98 | 46 | 52
ICD9CM | 43 | 31 | 12
ICDO3 | 229 | 5 | 224
Nebraska   Lexicon | 7 | 6 | 0
SNOMED | 847 | 847 | 0



**2) Counts source concepts, and the corresponding vocabulary mapping**
Takeaway: There is significant inconsistency as to which vocabularies these are mapped to. 

vocabulary_id | concept_count | map_to_snomed | map_to_cancermodifier | map_to_icdo3
-- | -- | -- | -- | --
ICD10CM | 59 | 46 | 52 | 0
ICD9CM | 31 | 31 | 12 | 0
ICDO3 | 229 | 0 | 224 | 5
Nebraska   Lexicon | 7 | 0 | 0 | 0
SNOMED | 847 | 847 | 0 | 0


Concept level mapping counts attached: 
[mets_source_mapping_compare.csv](https://github.com/OHDSI/OncologyWG/files/12374220/mets_source_mapping_compare.csv)

**Comment by cgreich** [comment]
@rtmill:

Not sure I follow. You may want to clarify.

What is it you want to "indicate"? An "indication" in medicine is a reason for treatment (or prevention). So, Tylenol is indicated for pain or fever.

How does that work with metastasis? Or are you trying to establish a link between the primary and a metastasis in the data? Or are you inferring a metastasis from a source code? If the latter, "Maps to" is all you need.
